src/BlockTypes/AddToCartForm.php

```diff
@@ -25,33 +25,28 @@ class AddToCartForm extends AbstractBlock {
 	 * @return string | void Rendered block output.
 	 */
 	protected function render( $attributes, $content, $block ) {
-		global $product;
-
 		$post_id = $block->context['postId'];
 
 		if ( ! isset( $post_id ) ) {
 			return '';
 		}
 
-		if ( ! $product instanceof \WC_Product ) {
-			$product = wc_get_product( $post_id );
-			if ( ! $product instanceof \WC_Product ) {
-				return '';
-			}
+		$single_product = wc_get_product( $post_id );
```

Comment:
The `single` prefix seems redundant here for the variable. The original variable name was fine, the main issue was the global declaration which meant the variable value was getting clobbered in the global scope. Without the declaration, you can use `$product`.

Reply:
> The original variable name was fine, the main issue was the global declaration which meant the variable value was getting clobbered in the global scope. Without the declaration, you can use $product.

Yes: I did this change on purpose to make it even **more** evident that we are working with a local variable, in practice it doesn't change anything indeed. I'm glad to change it back to just `$product` if `$single_product` doesn't sound intuitive.